main_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+var kernelImagePath = "./vmlinux"
+var rootDrivePath = "./root-drive.img"
+
+func init() {
+	if v := os.Getenv("KERNELIMAGE"); len(v) != 0 {
+		kernelImagePath = v
+	}
+
+	if v := os.Getenv("ROOTDRIVE"); len(v) != 0 {
+		rootDrivePath = v
+	}
+}
+
+func TestFireCTL(t *testing.T) {
+	const firectlName = "./firectl"
+	if _, err := os.Stat(firectlName); os.IsNotExist(err) {
+		t.Skip("Missing firectl binary")
+	}
+
+	if _, err := os.Stat(kernelImagePath); os.IsNotExist(err) {
+		t.Skip("Missing kernel image")
+	}
+
+	if _, err := os.Stat(rootDrivePath); os.IsNotExist(err) {
+		t.Skip("Missing root drive")
+	}
+
+	ctx := context.Background()
+	ctlCtx, cancelFn := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancelFn()
+
+	const socketpath = "./integration_test.sock"
+	firectlArgs := []string{
+		"-s",
+		socketpath,
+		"--kernel",
+		kernelImagePath,
+		"--root-drive",
+		rootDrivePath,
+	}
+	cmd := exec.CommandContext(ctlCtx, firectlName, firectlArgs...)
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("Failed to run command: %v", err)
+	}
+
+	defer func() {
+		if err := os.Remove(socketpath); err != nil {
+			t.Log("Failed ot remove socket path file")
+		}
+
+		if err := cmd.Process.Kill(); err != nil {
```

Comment:
Might as well send a `SIGQUIT`, after #24 We don't need to request a clean shutdown of this VM.

Reply:
Why not just rely on the `Kill` method? It sends a `SIGKILL` so I don't think we need to also send a `SIGQUIT`?

Reply:
We definitely don't want to send a `SIGKILL`. It will terminate the firectl process, but leave an orphaned firecracker process (granted, it should self-destruct once the kernel reboots, but still, we want to let firectl's signal handlers do their thing)

Reply:
@nmeyerhans - Ah, that makes sense! Cool, will go ahead and signal a `SIGQUIT` instead.